assets/javascripts/obs_rsync.js

```diff
@@ -0,0 +1,38 @@
+function postAndRefresh(btn, delay){
+    var post_url = $(btn).attr("post_url");
+    $.ajax({
+        url: post_url,
+        method: 'POST',
+        dataType: 'json',
+        success: function(data) {
+            if (delay) {
+                setTimeout(function() {
+                    location.reload();
+                }, delay);
+            } else {
+                location.reload();
+            }
+        },
+        error: function(xhr, ajaxOptions, thrownError) {
+            var message = xhr.responseJSON.error;
+            addFlash('danger', 'Error: ' + message);
+        }
+    });
+}
+
+
+function postAndRedirect(btn, redir){
```

Comment:
This function is redundant to `postAndRefresh` except for the success handler. It would make sense to our-source common parts.

Reply:
These functions are mess, I just want to achieve required functionality and don't feel much benefit from trying to find common abstractions here.

Reply:
somehow I feel this is not really an answer. However I leave it for @Martchus to follow on with this if he so desires

Reply:
I don't think it is hard to move common code into a separate function.